spaceR/spaceR/Controllers/SignupViewController.swift

```diff
@@ -9,8 +9,56 @@ import UIKit
 
 class SignupViewController: UIViewController {
     
+    @IBOutlet private weak var userFirstName: UITextField!
+    @IBOutlet private weak var userLastName: UITextField!
+    @IBOutlet private weak var userEmail: UITextField!
+    @IBOutlet private weak var userPassword: UITextField!
+    private lazy var signUpViewModel = SignupViewModel()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         self.navigationController?.navigationBar.isHidden = true
     }
+    
+    @IBAction private func signupButtonClicked(_ sender: Any) {
+        if self.validateUserInputFromSignupFields() {
+                 self.successfulSignup()
+             } else {
+                 self.unsuccessfullSignup()
+             }
+    }
+    
+    private func validateUserInputFromSignupFields() -> Bool {
+        if let safeFirstName = self.userFirstName.text, !safeFirstName.isEmpty,
+           let safeLastName = self.userLastName.text, !safeLastName.isEmpty,
+           let safeEmail = self.userEmail.text, !safeEmail.isEmpty,
+           let safePassword = self.userPassword.text,
+           !safePassword.isEmpty {
+            setSignupModelAttributes(firstName: safeFirstName, lastName: safeLastName, email: safeEmail, password: safePassword)
+            return true
+        } else {
+            self.displayErrorAlert(title: UIErrorAlerts.CustomError.unsuccessfulSignupDueToMisingFields.errorDescription!, errorMessage: UIErrorAlerts.CustomError.unsuccessfulSignupDueToMisingFields.failureReason!, buttonTitle: "Ok")
```

Comment:
So if you want to use your custom error `enum` you have to update this function to expect it 😄

Reply:
I tried but i got a bit confused doing it that way. Could you give me some guidance on how to approach doing that? The function expects 2 strings, which come from the enum (error message and failure message), but if i say CustomError.errorMessage, it doesn't find it. How would i change the function.

I tried displayErrorAlert(CustomFunction.type), but i need the errormessage and failure message that goes witht hat enum

Reply:
So instead of having that function expect a string have it expect one of your `enum` cases and then you can access the `rawValue` in the `displayErrorAlert` function.

Give me a call if you need some more info on this 😄